tools/check-symbol-names

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+[ -z "$@" ] && { echo "Usage: $0 libfoo.a" 1>&2; exit 2; }
+
+nm -A -P "$@" | awk '
+# ignore caml_foo, camlFoo_bar, and linker-internal _foo
+$2 ~ /^(caml[_A-Z]|_)/ { next }
```

Comment:
It would be nice if this script worked on the Mac as well.
At the moment I don't think it will, since all of the symbols on such platform start with an underscore...